LinearAlgebraForCAP/gap/DerivedMethodsForMatrixCategories.gi

```diff
@@ -0,0 +1,232 @@
+
+BindGlobal( "CATEGORY_FILTER_CHECKER",
+  function( cat )
+    local k, matrix_cat;
+    
+    # if the category is matrix category of some homalg field then do nothing
+    if IsBound( cat!.field_for_matrix_category ) then
+      
+      return false;
+      
+    fi;
+    
+    if not ( HasIsLinearCategoryOverCommutativeRing( cat ) and IsLinearCategoryOverCommutativeRing( cat ) ) then
+      
+      return false;
+      
+    fi;
+    
+    k := CommutativeRingOfLinearCategory( cat );
+    
+    if not ( HasIsFieldForHomalg( k ) and IsFieldForHomalg( k ) ) then
+      
+      return false;
+      
+    fi;
+    
+    matrix_cat := MatrixCategory( k );
+    
+    if not HasRangeCategoryOfHomomorphismStructure( cat ) then
+      
+      SetRangeCategoryOfHomomorphismStructure( cat, matrix_cat );
```

Comment:
This line is extremely dangerous.
You set an attribute within a function that is only supposed to check whether a certain derivation might be applicable. You cannot guarantee that even if the check yields true that the final derivation will be actually installed, and you would have set this attribute wrongly in that case.

I can think of two possible solutions:
1) we need to make a CAP primitive for `RangeCategoryOfHomomorphismStructure` and then you can derive this one as well
2) you bundle all the final derivations in this file to a "final derivation package" (like I suggested with the other final derivations above), and I provide a mechanism that calls a user given function (in which you could then set this attribute) in the case such a package is actually installed.

Can you think of other solutions? If no, maybe it's also not a bad idea to actually employ 1) and 2).

Reply:
To be done tomorrow.

Reply:
The idea of adding an optional function  to a "final derivation package" which is designed to be called before actually starting to add the components of the package, would be very helpful for future instances. So it would be very nice to have this mechanism.

The idea of adding a primitive operation which sets/computes the attribute "RangeCategoryOfHomomorphismStructure" is also feasible, exactly as if we are adding "ZeroObject" method; which is also an attribute to the category.

Reply:
In this commit https://github.com/homalg-project/CAP_project/pull/457/commits/c575c87c6b9efefab338dc7b3a6064e1a55976e5, I bundled the several final derivations to a single finale derivation package, and as soon as you implement the above or other mechanism, I will accordingly re-update the `CATEGORY_FILTER_CHECKER`.